lib/hls/hls_parser.js

```diff
@@ -1402,25 +1403,72 @@ shaka.hls.HlsParser = class {
     }
 
     let encrypted = false;
+    let clearKey;
     /** @type {!Array.<shaka.extern.DrmInfo>}*/
     const drmInfos = [];
     const keyIds = new Set();
 
-    // TODO: May still need changes to support key rotation.
+    /* eslint-disable no-await-in-loop */
+    // TODO: May still need changes to support key rotation and discontinuity.
     for (const drmTag of drmTags) {
       const method = drmTag.getRequiredAttrValue('METHOD');
-      if (method != 'NONE') {
+      if (method == 'AES-128') {
         encrypted = true;
+        this.aesEncrypted_ = true;
 
-        // We do not support AES-128 encryption with HLS yet. So, do not create
-        // StreamInfo for the playlist encrypted with AES-128.
-        // TODO: Remove the error message once we add support for AES-128.
-        if (method == 'AES-128') {
-          shaka.log.warning('Unsupported HLS Encryption', method);
-          this.aesEncrypted_ = true;
-          return null;
+        // Check if the Web Crypto API is available.
+        if (!window.crypto || !window.crypto.subtle) {
+          shaka.log.alwaysWarn('Web Crypto API is not available to decrypt ' +
+              'AES-128. (Web Crypto only exists in secure origins like https)');
+          throw new shaka.util.Error(
+              shaka.util.Error.Severity.CRITICAL,
+              shaka.util.Error.Category.MANIFEST,
+              shaka.util.Error.Code.HLS_KEYFORMATS_NOT_SUPPORTED);
+        }
+
+        // HLS RFC 8216 Section 5.2:
+        // An EXT-X-KEY tag with a KEYFORMAT of "identity" that does not have an
+        // IV attribute indicates that the Media Sequence Number is to be used
+        // as the IV when decrypting a Media Segment, by putting its big-endian
+        // binary representation into a 16-octet (128-bit) buffer and padding
+        // (on the left) with zeros.
+        let firstMediaSequenceNumber = 0;
+        let iv;
+        const ivHex = drmTag.getAttributeValue('IV', '');
+        if (!ivHex) {
+          // Media Sequence Number will be used as IV.
+          firstMediaSequenceNumber =
+              shaka.hls.Utils.getFirstTagWithNameAsNumber(
+                  playlist.tags, 'EXT-X-MEDIA-SEQUENCE', 0);
+        } else {
+          // Exclude 0x at the start of string.
+          iv = shaka.util.Uint8ArrayUtils.fromHex(ivHex.substr(2));
+          if (iv.byteLength != 16) {
+            throw new shaka.util.Error(
+                shaka.util.Error.Severity.CRITICAL,
+                shaka.util.Error.Category.MANIFEST,
+                shaka.util.Error.Code.HLS_KEYFORMATS_NOT_SUPPORTED);
```

Comment:
I think it is better to add a new error code for this case. Because then there is a better explanation of what is happening.

Reply:
Done.